.github/workflows/automation-scripts/update-changelogs.mjs

```diff
@@ -0,0 +1,120 @@
+* open PR into target branch
+*****************************************************************/
+
+const targetPath = "./temp-target-changelogs"
+const incomingPath = "./temp-incoming-changelogs"
+
+// To run shell commands using zx use "await $`cmd`"
+await $`mkdir ${targetPath}`
+await $`mkdir ${incomingPath}`
+
+// find the latest release branch, and make that the target for the changelogs
+let targetBranch = await $`git branch -a --list "origin/release/[0-9]*.[0-9]*.x" | tail -n1 | sed 's/  remotes\\///'`;
+let currentBranch = await $`git branch --show-current`;
+let commitMessage = await $`git log --format=%B -n 1`;
+
+
+targetBranch = String(targetBranch).slice(0, -1);
+currentBranch = String(currentBranch).slice(0, -1);
+commitMessage = String(commitMessage).slice(0, -1);
+
+if (targetBranch === `origin/${currentBranch}`) {
+  console.log("The current branch is the latest release, so the target will be master branch")
+  targetBranch = 'master'
+} else
+  console.log(`The current branch is ${currentBranch}, so the target will be ${targetBranch} branch`)
+
+// copy all changelogs from the current branch to ./temp-incoming-changelogs, the files will be named: package_name_CHANGELOG.json
+await $`find ./ -type f -name "CHANGELOG.json" -not -path "*/node_modules/*" -exec sh -c 'cp "{}" "./temp-incoming-changelogs/$(echo "{}" | sed "s/^.\\///; s/\\//_/g")"' \\;`;
+// # copy all changelogs from the target branch to ./temp-target-changelogs, the files will be named: package_name_CHANGELOG.json
+await $`git checkout ${targetBranch}`;
+await $`find ./ -type f -name "CHANGELOG.json" -not -path "*/node_modules/*" -exec sh -c 'cp "{}" "./temp-target-changelogs/$(echo "{}" | sed "s/^.\\///; s/\\//_/g")"' \\;`;
+
+const currentFiles = getFilePaths(targetPath);
+currentFiles.forEach((file, index) => {
+  currentFiles[index] = file.split('/').slice(1);
+})
+fixChangeLogs(currentFiles);
+
+// copy changelogs back to proper file paths and convert names back to: CHANGELOG.json
+await $`find ./temp-target-changelogs/ -type f -name "*CHANGELOG.json" -exec sh -c 'cp "{}" "$(echo "{}" | sed "s|temp-target-changelogs/\\(.*\\)_|./\\1/|; s|_|/|g")"' \\;`;
+// delete temps
+await $`rm -r ${targetPath}`;
+await $`rm -r ${incomingPath}`;
+// # regen CHANGELOG.md
+await $`rush publish --regenerate-changelogs`;
+/*********************************************************************/
+// Uncomment For Manual runs and fix branch name to appropriate version
+// the version should match your incoming branch
+// await $`git checkout -b finalize-release-X.X.X`;
+/*********************************************************************/
+await $`git add .`;
+await $`git commit -m "${commitMessage} Changelogs"`;
+await $`rush change --bulk --message "" --bump-type none`;
+await $`git add .`;
+await $`git commit --amend --no-edit`;
+await $`git push https://$(GITHUBTOKEN)@github.com/iTwin/itwinjs-core HEAD:${targetBranch}`
```

Comment:
These are the only lines i didn't test (67-70). I got a successful run in the workflow, but i had to modify the script to accommodate testing without the correct source and target branches. Assuming this git push command works as intended this should be full functional. I did a regular git push when tested and excluded the two args. I don't fully know if this cmd is correct, but it is how it was initially written for the original workflow, so that is what I'm basing it of off.

Reply:
I assumed it was a var stored in the workflow, but I can't find a secret vars section inside GH Actions. This was how the workflow was written before, but technically don't know if it ever worked since it would break before reaching this. I'll try to run a quick test

Reply:
Yeah i still don't really know where that var comes from but the cmd does work. I'll still try to figure that out though. And sorry wasn't thinking and accidentally requested core-admin review when making a dummy push